### PR TITLE
[api][SIMS #231] Enable Tests on Github Action CI/CD

### PR DESCRIPTION
**Pull request by fwpushan**
-----
1. Create new make rule to run api test on docker env
2. Update tests accordingly
3. Add new github test workflow
----
N.B. For Reviewers 
-------
**Currently most of the controllers and services required database connection, so to avoid multiple connection and speed up the test execution, change subject initialisation beforEach to beforAll**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          echo Git Build ID: ${{ github.event.number }}
+          echo Git Pull Request Ref: ${{ github.event.pull_request.head.sha }}
+  test:
+    name: Test API in Docker env
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.merged == false }}
+    env:
+      PROJECT_NAME: aest-sims
+      BUILD_ID: ${{ github.event.number }}
+      POSTGRES_PASSWORD: bcgov!
```
Comment (andrewsignori-aot) [inline .github/workflows/test.yml:25]: Even that this password is used only for the test DB, consider moving to a secret.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        uses: actions/checkout@v1
+      - name: Docker Pull
+        working-directory: "./sources/"
+        run: docker-compose pull
+      # In this step, this action saves a list of existing images,
+      # the cache is created without them in the post run.
+      # It also restores the cache if it exists.
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        # Ignore the failure of a step and avoid terminating the job.
+        continue-on-error: true
```
Comment (andrewsignori-aot) [inline .github/workflows/test.yml:59]: Could you please elaborate on why we should ignore the error here?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      # Deploy the api
+      - name: Build and run API in Docker
+        working-directory: "./sources/"
+        run: |
+          make local-api-detached
+      - name: Sleep for 10 seconds
```
Comment (andrewsignori-aot) [inline .github/workflows/test.yml:65]: Could you please elaborate on why we should wait for 10s here?

File: .github/workflows/test.yml (hunk 1 above)
Comment (fwpushan) [reply .github/workflows/test.yml:25]: Same info is already exposed in sources/make file as default value .....

File: .github/workflows/test.yml (hunk 2 above)
Comment (fwpushan) [reply .github/workflows/test.yml:59]: This is docker caching, this is to speedup docker build process, if that fails, still our tests run but with some extra time.

File: .github/workflows/test.yml (hunk 3 above)
Comment (fwpushan) [reply .github/workflows/test.yml:65]: Currently, booting up local db host taking sometime in Github action machine adding a little bit wait resolving that.

File: sources/packages/api/src/services/auth/keycloak/keycloak.service.ts [hunk 4]
```diff
@@ -110,6 +110,7 @@ export class KeycloakService {
     } catch (ex) {
       // TODO: Add a logger.
       this.logger.error(ex);
+      console.error(ex);
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/auth/keycloak/keycloak.service.ts:113]: I believe that you can remove it now.

**Review by 2011sandeepraj — COMMENTED** [review]
LGTM

File: sources/packages/api/test/auth.e2e-spec.ts [hunk 5]
```diff
@@ -11,6 +11,8 @@ import {
 import { AuthTestController } from "../src/testHelpers/controllers/auth-test/auth-test.controller";
 import { KeycloakService } from "../src/services/auth/keycloak/keycloak.service";
 
+jest.setTimeout(10000);
```
Comment (andrewsignori-aot) [inline sources/packages/api/test/auth.e2e-spec.ts:14]: Was it really necessary or was it more a local issue?

**Review by andrewsignori-aot — APPROVED** [review]
👍

**Pull request merged**